core/src/test/java/org/apache/iceberg/avro/AvroDataUnionRecordTest.java

```diff
@@ -0,0 +1,183 @@
+        "  \"type\": \"record\",\n" +
+        "  \"fields\": [\n" +
+        "    {\n" +
+        "      \"name\": \"map\",\n" +
+        "      \"type\": [\n" +
+        "        \"null\",\n" +
+        "        {\n" +
+        "          \"type\": \"map\",\n" +
+        "          \"values\": [\n" +
+        "            \"null\",\n" +
+        "            \"boolean\",\n" +
+        "            \"int\",\n" +
+        "            \"long\",\n" +
+        "            \"float\",\n" +
+        "            \"double\",\n" +
+        "            \"bytes\",\n" +
+        "            \"string\"\n" +
+        "          ]\n" +
+        "        }\n" +
+        "      ]\n" +
+        "    }\n" +
+        "  ]\n" +
+        "}";
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(schema1);
+    org.apache.iceberg.Schema icebergSchema = AvroSchemaUtil.toIceberg(avroSchema);
+    org.apache.avro.Schema avroSchemaUnionRecord = AvroSchemaUtil.convert(icebergSchema, "test");
+    org.apache.avro.Schema unionRecordSchema =
+        avroSchemaUnionRecord.getFields().get(0).schema().getTypes().get(1).getValueType().getTypes().get(1);
+
+    List<GenericData.Record> expectedRead = new ArrayList<>();
+    List<GenericData.Record> actualWrite = new ArrayList<>();
+
+    for (long i = 0; i < 10; i++) {
+      Map<String, Object> map = new HashMap<>();
+      Map<String, Object> mapRead = new HashMap<>();
+      updateMapsForUnionSchema(unionRecordSchema, map, mapRead, i);
+      GenericData.Record recordRead = new GenericRecordBuilder(avroSchema)
+          .set("map", mapRead)
+          .build();
+      GenericData.Record record = new GenericRecordBuilder(avroSchema)
+          .set("map", map)
+          .build();
+      actualWrite.add(record);
+      expectedRead.add(recordRead);
+    }
+    writeAndValidate(actualWrite, expectedRead, icebergSchema);
+  }
+
+  private void updateMapsForUnionSchema(
+      org.apache.avro.Schema unionRecordSchema,
+      Map<String, Object> map,
+      Map<String, Object> mapRead,
+      Long index) {
+    map.put("boolean", index % 2 == 0);
+    map.put("int", index.intValue());
+    map.put("long", index);
+    map.put("float", index.floatValue());
+    map.put("double", index.doubleValue());
+    map.put("bytes", ByteBuffer.wrap(("bytes_" + index).getBytes()));
+    map.put("string", "string_" + index);
+
+    map.entrySet().stream().forEach(e -> {
+      String key = e.getKey();
+      GenericData.Record record = getGenericRecordForUnionType(unionRecordSchema, map, key);
+      mapRead.put(key, record);
+    });
+  }
+
+  private GenericData.Record getGenericRecordForUnionType(
+      org.apache.avro.Schema unionRecordSchema,
+      Map<String, Object> map,
+      String key) {
+    GenericRecordBuilder rec = new GenericRecordBuilder(unionRecordSchema);
+    switch (key) {
+      case "boolean":
+        return rec
+            .set("member1", map.get(key))
```

Comment:
Is it possible to use "option_1" instead of "member1"? Avro refers to these as union options.

Reply:
ack